pkg/acsengine/engine.go

```diff
@@ -685,8 +685,28 @@ func getDCOSAgentProvisionScript(profile *api.AgentPoolProfile) string {
 		roleFileContents = "touch /etc/mesosphere/roles/slave"
 	}
 
-	provisionScript = strings.Replace(provisionScript, "ROLESFILECONTENTS", roleFileContents, -1)
+	//look for an agent specific script to add
+
+	as, err2 := Asset("dcosagentcustomscript.sh")
+
```

Comment:
Change it to be like this:

if as, err2 := Asset("dcosagentcustomscript.sh"); err2 != nil {
    panic(fmt.Sprintf("BUG: %s", err2.Error()))
}

// then the rest of the code here.